interceptors.go

```diff
@@ -44,30 +50,64 @@ func FilterMethodsFunc(ctx context.Context, fullMethodName string) bool {
 	return true
 }
 
-//SetFilterFunc sets the default filter function to be used by interceptors
+// SetFilterFunc sets the default filter function to be used by interceptors
 func SetFilterFunc(ctx context.Context, ff FilterFunc) {
 	if ff != nil {
 		defaultFilterFunc = ff
 	}
 }
 
+// AddUnaryServerInterceptor adds a server interceptor to default server interceptors
+func AddUnaryServerInterceptor(ctx context.Context, i grpc.UnaryServerInterceptor) {
+	unaryServerInterceptors = append(unaryServerInterceptors, i)
+}
+
+// AddStreamServerInterceptor adds a server interceptor to default server interceptors
+func AddStreamServerInterceptor(ctx context.Context, i grpc.StreamServerInterceptor) {
+	streamServerInterceptors = append(streamServerInterceptors, i)
+}
+
+// ReplaceColdBrewServerInterceptors allows enabling/disabling coldbrew server interceptors
+//
+// when set to true, the coldbrew server interceptors will not used
+func ReplaceColdBrewServerInterceptors(ctx context.Context, flag bool) {
```

Comment:
Thanks for this change @ankurs
Suggestion: Can we consider making the new Add methods variadic and accept a list of interceptors? Clients can set DefaultInterceptors and/or their custom interceptors and we can avoid the replaceServerInterceptors flag. i.e. all desired interceptors, whether default or custom, are added explicitly by clients and coldbrew uses only the added interceptors.
Advantage: Behaviour becomes explicitly opt-in from clients and we do not need the replace logic.

However this is a !! BREAKING !! change for clients who upgrade coldbrew expecting the defaults to work out of the box. Not sure if that is acceptable.

Reply:
I have made the functions variadic, typically we don't expect majority of services to need custom interceptors and we just provide a sane default for them to work with

on the default interceptors I thinks its better to make them opt-in, with the ability to disable them which also preserves the current behaviour

Reply:
Makes sense 👍